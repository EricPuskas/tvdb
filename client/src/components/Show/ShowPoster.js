import React from "react";

// Utilities
import replaceImage from "../../utils/replaceImage";

// Components
import NoPosterFound from "../../assets/img/no_poster.png";

const ShowPoster = ({ poster, title }) => {
  return (
    <div className="ShowPoster" data-test="poster">
      <img
        onError={e => replaceImage(e, NoPosterFound)}
        src={
          poster ? `https://www.thetvdb.com/banners/${poster}` : NoPosterFound
        }
        alt={title}
      />
    </div>
  );
};

export default ShowPoster;
