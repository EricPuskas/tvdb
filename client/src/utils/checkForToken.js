import jwt_decode from "jwt-decode";

// Redux Store
import store from "../store";

// Utilities
import setToken from "./setToken";

// Actions
import { getToken } from "../actions/token";

const checkForToken = () => {
  // Check for token
  if (!localStorage.token) {
    store.dispatch(getToken());
    setToken(localStorage.token);
  } else {
    setToken(localStorage.token);
    // Decode token to be able to check if it expired
    const decoded = jwt_decode(localStorage.token);
    // Check for expired token
    const currentTime = Date.now() / 1000;
    if (decoded.exp < currentTime) {
      store.dispatch(getToken());
    }
  }
};

export default checkForToken;
