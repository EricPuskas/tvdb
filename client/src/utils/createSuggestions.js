const filterInput = input => {
  return input.replace(/[^\w\s]/gi, "");
};

const createSuggestions = (word, data) => {
  let filter = filterInput(word);
  const re = new RegExp(`${filter.toLowerCase()}.*\\B`, "g");
  return data.filter(item => re.test(item.toLowerCase()));
};

export default createSuggestions;
