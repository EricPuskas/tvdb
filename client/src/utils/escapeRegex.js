const escapeRegex = text => {
  if (text !== "" || text !== null || text !== undefined) {
    // replace all white space with +, globally (replace all occurrences)
    return text.replace(/[\s]/g, "+");
  }
};

export default escapeRegex;
