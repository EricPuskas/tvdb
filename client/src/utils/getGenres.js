import React from "react";
import uuid from "uuid/v4";

const getGenres = genres => {
  if (!genres) return null;
  return genres.map(genre => (
    <li key={uuid()}>
      <span className="tag">{genre}</span>
    </li>
  ));
};

export default getGenres;
