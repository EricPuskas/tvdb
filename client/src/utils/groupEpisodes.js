const groupEpisodes = data => {
  let seasons = [];
  for (let i = 0; i < data.length; i++) {
    if (!seasons[data[i].airedSeason]) seasons[data[i].airedSeason] = [];
    seasons[data[i].airedSeason].push({
      id: data[i].id,
      season: data[i].airedSeason,
      episode: data[i].airedEpisodeNumber,
      title: data[i].episodeName
    });
  }
  return seasons;
};

export default groupEpisodes;
