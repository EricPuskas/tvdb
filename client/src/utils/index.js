import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
