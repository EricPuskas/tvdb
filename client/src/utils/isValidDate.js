import moment from "moment";

const isValidDate = date => {
  return moment(date).isValid();
};

export default isValidDate;
