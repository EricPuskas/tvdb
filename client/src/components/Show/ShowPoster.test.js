import React from "react";
import { shallow } from "enzyme";
import { findByAttribute } from "../../utils";
import ShowPoster from "./ShowPoster";
import NoPosterFound from "../../assets/img/no_poster.png";

describe("ShowPoster Component", () => {
  const component = shallow(<ShowPoster />);
  const wrapper = findByAttribute(component, "poster");
  const props = {
    poster: "graphical/257655-g9.jpg",
    title: "Some Title"
  };
  it("Should render without errors", () => {
    expect(wrapper.length).toBe(1);
  });

  it("Render placeholder image", () => {
    const logo = shallow(<ShowPoster />);

    expect(logo.find("img").prop("src")).toEqual(NoPosterFound);
  });

  it("Render banner image", () => {
    const logo = shallow(<ShowPoster {...props} />);

    expect(logo.find("img").prop("src")).toEqual(
      "https://www.thetvdb.com/banners/" + props.poster
    );
  });

  it("Should pass snapshot test", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
