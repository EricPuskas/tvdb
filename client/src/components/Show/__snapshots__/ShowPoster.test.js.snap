// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPoster Component Should pass snapshot test 1`] = `"<div class=\\"ShowPoster\\" data-test=\\"poster\\"><img src=\\"no_poster.png\\"/></div>"`;
